src/lib/Preview.js

```diff
@@ -462,13 +467,25 @@ class Preview extends EventEmitter {
      */
     download() {
         const { apiHost, queryParams } = this.options;
+        let downloadUrl;
 
         if (checkPermission(this.file, PERMISSION_DOWNLOAD)) {
             // Append optional query params
-            const downloadUrl = appendQueryParams(getDownloadURL(this.file.id, apiHost), queryParams);
+            downloadUrl = appendQueryParams(getDownloadURL(this.file.id, apiHost), queryParams);
             get(downloadUrl, this.getRequestHeaders()).then((data) => {
                 openUrlInsideIframe(data.download_url);
             });
+        } else if (canDownloadWatermarkedRep(this.file)) {
+            // If we can download the watermarked rep but not the original, we assume that the viewer is using the
+            // watermarked rep for preview, so we will download it.
```

Comment:
I like putting the comment for the block above so you can separate the comments for the block from the comment for the first line, like:

```
    get(downloadUrl, this.getRequestHeaders()).then((data) => {
        openUrlInsideIframe(data.download_url);
    });
    
    // If we can download the watermarked rep but not the original, we assume that the viewer is using the
    // watermarked rep for preview, so we will download it.
} else if (canDownloadWatermarkedRep(this.file)) {
    // This allows the content URL to be downloaded by the browser
    const downloadQueryParams = { response_content_disposition_type: 'attachment' };
```